packages/cli-kit/src/public/node/base-command.ts

```diff
@@ -28,7 +28,7 @@ abstract class BaseCommand extends Command {
   }
 
   async catch(error: Error & {exitCode?: number | undefined}): Promise<void> {
-    errorHandler(error, this.config)
+    await errorHandler(error, this.config)
```

Comment:
I don't think the overload signatures are helpful (since we're not using return types) but I also don't think we'd ever get an error on this. Why should TS care if it's awaited? You're always free to just fire-and-forget an async function, and we do that elsewhere. As long as you don't try and use the return value assuming it's not a Promise, you'll be fine.

Reply:
Great, I removed the unnecessary overload signature and tightened up the return type to be either a Promise or unknown. Let me know if you have any suggestions for simplifying the return type further or any other ideas!